clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -899,6 +899,11 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (CodeGenOpts.PointerAuth.IndirectGotos)
     Fn->addFnAttr("ptrauth-indirect-gotos");
 
+  // Add return control flow integrity attributes for RISCV.
+  if (CodeGenOpts.CFProtectionReturn &&
```

Comment:
We already set a module flag cf-protection-return; why do we need both a module flag and a function flag?

If we do need something RISCV-specific here, can we do it in setTargetAttributes, or something like that?

Reply:
`cf-protection-return` on RISC-V could be controlled at a per-function granularity (i.e. use function attributes to suppress the instruction generation for some functions), so we still need a function attribute to tag it.
I moved it to `setTargetAttributes` for RISC-V, thanks for the advice.